gokart/mypy.py

```diff
@@ -0,0 +1,373 @@
+                # set on self in the generated __init__(), not in the class body. On the other
+                # hand, we don't know how custom task_on_kart transforms initialize attributes,
+                # so we don't treat them as implicit. This is required to support descriptors
+                # (https://github.com/python/mypy/issues/14868).
+                sym.implicit = True
+
+            current_attr_names.add(lhs.name)
+            with state.strict_optional_set(self._api.options.strict_optional):
+                init_type = self._infer_task_on_kart_attr_init_type(sym, lhs.name, stmt)
+            found_attrs[lhs.name] = TaskOnKartAttribute(
+                name=lhs.name,
+                has_default=has_default,
+                line=stmt.line,
+                column=stmt.column,
+                type=init_type,
+                info=cls.info,
+                api=self._api,
+            )
+        return list(found_attrs.values())
+
+    def _collect_parameter_args(self, expr: Expression) -> tuple[bool, dict[str, Expression]]:
+        """Returns a tuple where the first value represents whether or not
+        the expression is a call to luigi.Parameter() or gokart.TaskInstanceParameter()
+        and the second value is a dictionary of the keyword arguments that luigi.Parameter() or gokart.TaskInstanceParameter() was called with.
+        """
+        if isinstance(expr, CallExpr) and isinstance(expr.callee, RefExpr):
+            args = {}
+            for name, arg in zip(expr.arg_names, expr.args):
+                if name is None:
+                    # NOTE: this is a workaround to get default value from a parameter
+                    self._api.fail(
+                        'Positional arguments are not allowed for parameters when using the mypy plugin. '
+                        "Update your code to use named arguments, like luigi.Parameter(default='foo') instead of luigi.Parameter('foo')",
+                        expr,
+                    )
+                    continue
+                args[name] = arg
+            return True, args
+        return False, {}
+
+    def _infer_task_on_kart_attr_init_type(self, sym: SymbolTableNode, name: str, context: Context) -> Type | None:
+        """Infer __init__ argument type for an attribute.
+
+        In particular, possibly use the signature of __set__.
+        """
+        default = sym.type
+        if sym.implicit:
+            return default
+        t = get_proper_type(sym.type)
+
+        # Perform a simple-minded inference from the signature of __set__, if present.
+        # We can't use mypy.checkmember here, since this plugin runs before type checking.
+        # We only support some basic scanerios here, which is hopefully sufficient for
+        # the vast majority of use cases.
+        if not isinstance(t, Instance):
+            return default
+        setter = t.type.get('__set__')
+        if setter:
```

Comment:
it seems to be prefer to return early.

```python
if not setter:
    return default
```

Reply:
[83e1f53](https://github.com/m3dev/gokart/pull/384/commits/83e1f531831ad119f6ea0a8cfa9c7bfb140821cc)